docs/guide/api-javascript.md

```diff
@@ -200,3 +200,7 @@ async function transformWithEsbuild(
   inMap?: object
 ): Promise<ESBuildTransformResult>
 ```
+
+## Usage Notes
+
+When using `createServer` and `build` in the same Node.js process, both functions rely on `process.env.NODE_ENV` to work properly, which also depends on the `mode` config option. To prevent conflicting behaviour, set `process.env.NODE_ENV` or the `mode` of the two APIs to `development`. Otherwise, you can spawn a child process to run the APIs separately.
```

Comment:
Maybe add it as a note in `createServer`? I know that it should be in both and that is why it is in a new section here, but I think making the note be close to one of them would increase the chances of people seeing it.

Reply:
updated